sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -81,9 +81,35 @@ TEST(Context, BasicChar)
   EXPECT_TRUE(kind == ContextValue::ContextValueType::StdString);
 }
 
+TEST(Context, ApplicationContext)
+{
+  Context appContext = GetApplicationContext();
+
+  EXPECT_FALSE(appContext.HasKey("Key"));
+  EXPECT_FALSE(appContext.HasKey("key"));
+  EXPECT_FALSE(appContext.HasKey("Value"));
+  EXPECT_FALSE(appContext.HasKey("value"));
+  EXPECT_FALSE(appContext.HasKey("1"));
+  EXPECT_FALSE(appContext.HasKey(""));
+
+  auto duration = std::chrono::milliseconds(500);
+  EXPECT_FALSE(appContext.IsCancelled());
+  std::this_thread::sleep_for(duration);
+  EXPECT_FALSE(appContext.IsCancelled());
+
+  appContext.Cancel();
+  EXPECT_TRUE(appContext.IsCancelled());
+
+  //AppContext2 is the same context as AppContext
+  //  The context should be cancelled
+  Context appContext2 = GetApplicationContext();
+  EXPECT_TRUE(appContext.IsCancelled());
+
+}
+
 TEST(Context, IsCancelled)
 {
-  auto duration = std::chrono::milliseconds(150);
+  auto duration = std::chrono::milliseconds(500);
```

Comment:
Why change this value?

Reply:
150 feels quick and at risk at creating a flaky test if the hardware is running a bit slow.  I'll adjust to 250 to give a bit more wiggle room

Reply:
150 ms for computers is slow? :p
That's plenty of time to run a whole bunch of instructions and these tests are probably doing `< 10 ms` of work.

I don't think we ever saw such test fail because of timing flakiness, but your call. 250 is fine too.